protobuf/health.proto

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+syntax = "proto3";
+
+package grpc.health.v1;
+
+//@@.. cpp:namespace:: grpc.health.v1
+
+//@@
+//@@.. cpp:var:: message HealthCheckRequest
+//@@
+//@@   Request message for HealthCheck
+//@@
+message HealthCheckRequest
+{
+  string service = 1;
+}
+
+//@@
+//@@.. cpp:var:: message HealthCheckResponse
+//@@
+//@@   Response message for HealthCheck
+//@@
+message HealthCheckResponse
+{
+  //@@
+  //@@.. cpp:enum:: ServingStatus
+  //@@
+  //@@   Statuses supported by GRPC's health check.
+  //@@
+  enum ServingStatus {
+    UNKNOWN = 0;
+    SERVING = 1;
+    NOT_SERVING = 2;
+    SERVICE_UNKNOWN = 3;  // Used only by the Watch method.
+  }
+  ServingStatus status = 1;
+}
+
+//@@
+//@@.. cpp:var:: service Health
+//@@
+//@@   Health service for GRPC endpoints.
+//@@
+service Health
+{
+  //@@  .. cpp:var:: rpc Check(HealthCheckRequest) returns
+  //@@       (HealthCheckResponse)
+  //@@
+  //@@     Get serving status of the inference server.
+  //@@
+  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
+
+  //@@  .. cpp:var:: rpc Watch(HealthCheckRequest) returns
+  //@@       (stream HealthCheckResponse)
+  //@@
+  //@@     Get current serving status of the inference server and
```

Comment:
What do you mean by future status changes? Can you elaborate about `Watch()` function? How does `SERVICE_UNKNOWN` used in context of `Watch()`?

I wonder if the function is not implemented, then why should it be included in the RPC proto.

Reply:
I'll remove it from the RPC proto. I was trying to keep the original protobuf intact, as described in the [GRPC health checking documentation](https://github.com/grpc/grpc/blob/master/doc/health-checking.md), and leave it for users to see. Plus, it'd make implementing it in the future easier if we decide to. But you're right, we can remove it from the proto.

In case you're still curious about Watch, the official protobuf actually has more detailed comments [here](https://github.com/grpc/grpc/blob/master/src/proto/grpc/health/v1/health.proto#L47-L62) that we could comment if we choose to implement it in the future.

Reply:
Done. Let me know if that works for you, @tanmayv25.